### PR TITLE
[12.0][IMP] agreement_legal: add custom dynamic parties

### PR DESCRIPTION
**Pull request by ygol**
Add the option for the user to replace the default  parties section by a dynamic custom html field.

### CODE REVIEW

File: agreement_legal/models/agreement.py [hunk 1]
```diff
@@ -170,6 +170,20 @@ def _default_stage_id(self):
     company_contact_email = fields.Char(
         related="company_contact_id.email", string="Email"
     )
+    use_parties_content = fields.Boolean(
+        string="Use parties content",
+        default = False,
```
Comment (max3903) [inline agreement_legal/models/agreement.py]:
```suggestion
        default=False,
```

File: agreement_legal/models/agreement.py [hunk 2]
```diff
@@ -170,6 +170,20 @@ def _default_stage_id(self):
...
+        help="Use custom content for parties" )
```
Comment (max3903) [inline agreement_legal/models/agreement.py]:
```suggestion
        help="Use custom content for parties")
```

**Review by max3903 — APPROVED** [review]
LGTM. No test.

File: agreement_legal/models/agreement.py (hunk 1 above)
Comment (nikul-serpentcs) [reply agreement_legal/models/agreement.py]: no need to default False, Boolean always False

File: agreement_legal/models/agreement.py [hunk 3]
```diff
@@ -170,6 +170,19 @@ def _default_stage_id(self):
     company_contact_email = fields.Char(
         related="company_contact_id.email", string="Email"
     )
+    use_parties_content = fields.Boolean(
+        string="Use parties content",
+        help="Use custom content for parties")
+    parties = fields.Html(
+        string="Parties",
+        track_visibility="onchange",
+        help="Parties of the agreement",
```
Comment (max3903) [inline agreement_legal/models/agreement.py:206]: @ygol Can you add a default value mixing the current content with the dynamic fields? User would have a base to start with and no need to figure out the dynamic fields... Thank you.

Comment (ygol) [reply agreement_legal/models/agreement.py:206]: @max3903 Do you mean removing the boolean field and always show the html field with default value? 
Isn'n it cleaner to have both? 
IMO the form is much less cluttered without the html field (that is shown only if needed).

Comment (max3903) [reply agreement_legal/models/agreement.py:206]: We can keep the boolean field. I just think we should help the user with a default value.

**Review by osi-scampbell — APPROVED** [review]

**Review by max3903 — APPROVED** [review]

**Pull request merged**